### PR TITLE
Fixed #27421 -- Added shape, size, and offset controls to GDALRaster constructor.

### PR DESCRIPTION
**Pull request by yellowcap**
https://code.djangoproject.com/ticket/27421

### CODE REVIEW

File: docs/ref/contrib/gis/gdal.txt [hunk 1]
```diff
@@ -1117,16 +1117,39 @@ blue.
...
-        (1, 2)
+        (4, 4)
         >>> rst.bands[0].data()
-        array([[0, 1]], dtype=int8)
+        array([[5, 5, 5, 5],
+               [5, 2, 3, 5],
+               [5, 2, 3, 5],
+               [5, 5, 5, 5]], dtype=uint8)
+
+    .. versionchanged:: 1.11
```
Comment (timgraham) [inline docs/ref/contrib/gis/gdal.txt]: This note lacks some precision about what exactly is new. Is the entire note meant to be deleted when the versionchanged annotation is removed? If not, try to structure the note so it's self-contained as described in [Documenting new features](https://docs.djangoproject.com/en/dev/internals/contributing/writing-documentation/#documenting-new-features).

File: docs/ref/contrib/gis/gdal.txt [hunk 2]
```diff
@@ -1117,16 +1117,39 @@ blue.
...
+
+        The ``size``, ``shape``, and ``offset`` parameters can be specified
+        when creating a :class:`GDALRaster` object. The parameters can be
+        passed through the ds_input dictionary. This adds the ability to
```
Comment (timgraham) [inline docs/ref/contrib/gis/gdal.txt]: Add backticks around ds_input to fix docs build.

File: docs/ref/contrib/gis/gdal.txt (hunk 1 above)
Comment (yellowcap) [inline docs/ref/contrib/gis/gdal.txt]: The entire note can be removed with the versionchanged annotation. I think the complexity of the `ds_input` argument has grown and would benefit from its own section in the docs. I can start work on this. Do you think adding another subsection to the  [Raster Data Objects](https://docs.djangoproject.com/en/1.10/ref/contrib/gis/gdal/#raster-data-objects) would be appropriate?

Comment (timgraham) [inline docs/ref/contrib/gis/gdal.txt]: Sounds good.

**Pull request merged**